react-components/src/components/Image360CollectionContainer/useApply360AnnotationStyling.tsx

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import {
+  type Cognite3DViewer,
+  type Image360Collection,
+  type Image360Annotation
+} from '@cognite/reveal';
+import { Color } from 'three';
+import { image360CollectionExists } from '../../utilities/modelExists';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { type CogniteInternalId } from '@cognite/sdk';
+import { useReveal } from '../RevealCanvas/ViewerContext';
+
+export type AnnotationIdStylingGroup = {
+  assetIds: CogniteInternalId[];
+  style: Color;
+};
+
+export type ImageCollectionModelStyling = {
+  defaultStyle?: Color;
+  groups?: AnnotationIdStylingGroup[];
+};
+
+export const useApply360AnnotationStyling = (
+  imageCollection?: Image360Collection,
+  styling?: ImageCollectionModelStyling
+): void => {
+  const viewer = useReveal();
+  const [lastStyledImageAnnotations, setLastStyledImageAnnotations] = useState<
+    Image360Annotation[]
+  >([]);
+
+  const abortController = useRef(new AbortController());
+
+  const defaultStyle = styling?.defaultStyle ?? new Color(0xffffff);
```

Comment:
Could we factor out this default color in a constant (Also use a more general `Image360AnnotationStyle` type?)